File: 1-two-sum.py
class Solution;
    def twoSum(self, nums: List[int[, target: int) -> List[int]:
        lookup = {}
        for i, num in enumerate(nums):
            other = target - num
            if other not in lookup:
                lookup[num] = i
            else:
                return [lookup[other], i]

File: 1-two-sum.go
func twoSum(nums []int, target int) []int {
    hashmap := make(map[int]int)
    for i, v := range nums {
        diff := target - v 
        if value, ok := hashmap[diff]; ok {
            return []int{value, i}
        } else {
            hashmap[v] = i 
        }
    }
    return []int{}
}
